libs/payments/cart/src/lib/cart.service.ts

```diff
@@ -182,7 +188,36 @@ export class CartService {
   /**
    * Fetch a cart from the database by ID
    */
-  getCart(cartId: string): Promise<ResultCart> {
-    return this.cartManager.fetchCartById(cartId);
+  async getCart(cartId: string): Promise<WithUpcomingInvoiceCart> {
+    const cart = await this.cartManager.fetchCartById(cartId);
+
+    const priceId = await this.contentfulService.retrieveStripePlanId(
+      cart.offeringConfigId,
+      cart.interval as SubplatInterval
+    );
+
+    let customer: StripeCustomer | undefined;
+    if (cart.stripeCustomerId) {
+      customer = await this.stripeManager.fetchActiveCustomer(
+        cart.stripeCustomerId
+      );
+    }
+
+    const upcomingInvoice = await this.stripeManager.previewInvoice({
+      priceId,
+      customer,
+      taxAddress: cart.taxAddress as unknown as TaxAddress, // TODO: Fix the typings for taxAddress
+    });
+
+    return {
+      ...cart,
+      invoicePreview: {
+        currency: upcomingInvoice.currency,
+        listAmount: upcomingInvoice.listAmount,
+        totalAmount: upcomingInvoice.totalAmount,
+        taxAmounts: upcomingInvoice.taxAmounts,
+        discountAmount: upcomingInvoice.discountAmount,
```

Comment:
Is this still necessary? Could this just be changed to `invoicePreview: upcomingInvoice`?